framework/src/problems/FEProblemBase.C

```diff
@@ -221,6 +204,12 @@ FEProblemBase::validParams()
            "be used (again, using the parameter '" +
            list_param_name + "').";
   };
+
+  params.addParam<std::vector<SubdomainName>>(
+      "block",
+      {},
+      "Default list of subdomains for block-restrictable objects such as kernels and materials.");
```

Comment:
Can't we just delete this parameter? See my concern in https://github.com/idaholab/moose/pull/30630#discussion_r2129265023. Then `isParamSetByUser` is not even relevant and we don't need the new `isBlockSetByUserOrGlobalParams()` method. The only relevant thing then is `_has_block_in_global_params`

Reply:
As an aside I do believe `isParamSetByUser` would return true with the current code even if `block` is set in `GlobalParams` and not the `Problem` block itself

Reply:
> Can't we just delete this parameter? See my concern in [#30630 (comment)](https://github.com/idaholab/moose/pull/30630#discussion_r2129265023). Then `isParamSetByUser` is not even relevant and we don't need the new `isBlockSetByUserOrGlobalParams()` method. The only relevant thing then is `_has_block_in_global_params`

Agreed.

Reply:
Check out `CreateProblemDefaultAction`. That action is not tied to any input syntax, so I suppose the builder doesn't end up applying the global params to it?

Reply:
Yeah, found the fix -- thanks

Reply:
see https://github.com/idaholab/moose/pull/30300/commits/61438424ba1ee6397f76390f0e7da99343f15e79

Reply:
@lindsayad so after fixing `CreateProblemDefaultAction`, things are working correctly. We ended up keeping `Problem/block` but removing the `_has_block_in_global_params` since that's redundant. We need `Problem/block` because that's used as the list of blocks for coverage check.